rfcs/0122-system-permission.md

```diff
@@ -0,0 +1,712 @@
+| `api/workspaces/`  | `POST`  | `Write`  | _**Users can only create shared workspaces or personally owned ones. The additional `Project` permission is required to create protected workspaces.**_ |  |
+| `api/workspaces/{id}`  | `DELETE`  | `Write`  | _**Users can only remove shared workspaces and those that they own. The additional `Project` permission is required to remove protected workspaces.**_ |  |
+| `api/workspaces/{id}`  | `GET`  | `Read`  | Users can only read shared workspaces, and those that they own. |  |
+| `api/workspaces/{id}`  | `PUT`  | `Write`  | _**Users can only update shared workspaces and those that they own. The additional `Project` permission is required to update protected workspaces.**_ |  |
+| `api/workspaces/resources`  | `GET`  | `Public`  |  |  |
+| `api/workspaces/template`  | `GET`  | `Write`  |  |  |
+
+### `health/` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `health/`  | `GET`  | `Public`  |  |  |
+
+### `integrated/` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `integrated/`  | `GET`  | `Public`  | The Windows identity must already be authenticated. |  |
+
+### `oidc/challenge` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `oidc/challenge`  | `GET`  | `Public`  |  |  |
+| `oidc/challenge`  | `POST`  | `Public`  |  |  |
+
+### `theme/styles.css` 
+
+| Path | HTTP method | Permission demand | Additional requirements | Notes |
+| --- | --- | --- | --- | --- |
+| `theme/styles.css`  | `GET`  | `Public`  |  |  |
+
+## Backwards Compatibility
+
+To be impacted by this change an installation mostly needs to opt-in by allocating some users to the new `Project Owner` role. Existing `Administrators` will get the new `Project` permission and so will not notice a change. 
+
+API keys are created, via the UI and API, with a subset of the set of permissions. Thus any existing processes that created API keys with the `Setup` permission should be granted the `Project` and `System` permissions to preserve existing behaviour. This is valid because `Setup` is equal to `Project` + `System`.
+
+## UI
+
+### API Keys
+
+The API keys create and edit views will need to be changed to show the set of possible permissions as 'Read', 'Write', 'Ingest', 'Project', 'System'. 
+
+### Users
+
+The add/edit user and new user defaults views will need to reflect the additional user role `Project Owner`. 
+
+## Seq API
+
+[Seq API](https://github.com/datalust/seq-api) will need to be updated with the new permission and role.
+
+
+## Seq CLI
+
+[Seq CLI](https://github.com/datalust/seqcli) will need to be updated with the new permission and role. Possible just a documentation change. 
+
+# Future Directions
+
+NA.
+
+# Drawbacks
```

Comment:
"Drawbacks and Limitations"?

Does a discussion of app packages vs app instance permissions belong here? I.e. some apps like JSON Archive include some local system access (creating archive files); the change as proposed will still allow users with the `Project` permission to instantiate the JSON archive app and write to local files, if a user with `System` permission installs it.

Separating app instances into "Configurations" behind the `System` permission (assigning values to app settings, marking some as user-overridable), and "Outputs" under the `Write` permission (alongside notifications), will address this, eventually, but that change is still some way off.